### DISCUSSION TITLE
Auto-Switch C-, S-... Mode

### DISCUSSION THREAD

**Discussion by alexmaria8** [wmbusmeters/wmbusmeters#923] [Ideas]
Hey!

I'm using Wmbusmeters HA addon with a iM871a USB adapter on a Raspberry Pi.
I'm able to get the value of a water sensor (Qundis q water 5.5) in c-mode and the values ​​of few heatcostallocators (Qundis q caloric 5 AMR [not 5.5]) in s-mode.
Because i can't use C and S mode at the same time with this USB adapter, i wanna ask you if it's maybe possible to add a feature. It should switch from one mode to the other mode (and back) automatically for a defined time/interval without editing the config manually.

Best regards

**Comment by BIBOLV** [comment]
Hi,
It has been discussed in past and IMO it would come with lot of unnecessary code and config complexity to do that. There are two options now:
1. Get second dongle - then you will be able to use standard service and config
2. Create some simple script that would make switching for you. You can use existing config parameters, like `oneshot` and/or `exitafter` for that

**Reply by alexmaria8** [reply]
THX for your reply.
1. I have no more free usb-ports on my pi... maybe I should switch to a nuc or something similar but for now the pi was enough + this usb-adapter is not really cheap (50€+, but works fine)
2. For now I don't really know where to start (which program i have to use or is a simple texteditor enough, where to save the file, which fileformat i have to use, how to call it...)
But I'll check this out and try. Thank you for your answer!

**Reply by BIBOLV** [reply]
If you do not have experience with bash scripting, then it could be a bit challenging, so I quickly wrote for you place to start, but warning - this is totally untested, so use on your own risk 😉 . You can run wmbusmeters with all cfg provided as parameters or run against two different directories that contains configs, IMO later is more convenient.

```bash
#!/bin/bash

process_name="wmbusmeters"
value=1

while true; do
    if pgrep "$process_name" >/dev/null; then
        echo "Process $process_name is running"
    else
        echo -n "Process $process_name is not running, starting "
        if [[ "$value" -eq 1 ]]; then
            echo "in C mode"
	    wmbusmeters --useconfig=/etc/cmode &
	    # or wmbusmeters im871a:c1 ..... &
            value=2
        else
	    echo "in S mode"
	    wmbusmeters --useconfig=/etc/smode &
            value=1
        fi
    fi
    sleep 60
done
```

**Reply by alexmaria8** [reply]
Alright thank you! 
I'll do some backups and go into it. 🙂
